[HLSTree] Fix impaired detection for subtitles

CastagnaIT opened the pull request:
## Description
<!--- Provide a general summary of your change in the Pull Request title above -->
<!--- Describe your change in detail here. -->
Based on shaka player PR https://github.com/shaka-project/shaka-player/pull/5819
confirmed similar behaviour on D+ video service

## Motivation and context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
From discussion https://github.com/xbmc/xbmc/pull/25762#issuecomment-3323126629

CC @matthuisman

## How has this been tested?
<!--- Please describe in detail how you tested your change. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the Wiki documentation
- [ ] I have updated the documentation accordingly

matthuisman commented:
I'll test tonight

matthuisman commented:
Works great :) 
confirmed streams no longer marked as caption

thank you

matthuisman commented:
Do you think the check should also work if only public.accessibility.describes-music-and-sound is present?
I feel that should be marked as caption

So, the logic would simply be

    if (STRING::Contains(r.m_characteristics, "public.accessibility.describes-music-and-sound") ||
        STRING::Contains(r.m_characteristics, "public.accessibility.describes-video"))

CastagnaIT committed:
[HLSTree] Fix impaired detection for subtitles

Based on shaka player PR https://github.com/shaka-project/shaka-player/pull/5819
confirmed similar behaviour on D+ video service
